### ISSUE TITLE
need reverse machanism for Hazelcast's queue 

### ISSUE THREAD

**Issue by ppsnumi** [hazelcast/hazelcast#16586]
I use hazelcast's queue but inconvenience that get data from queue.

receiver:
```
loop(infinite) {
     data = queue.poll(5, TimeUnit.SECONDS);
     some delay logic
}
```

developer inconvenience get queue data.
developer does not need infinite loop and some delay logic(ex: Thread.sleep(1))
if queue have data or put data that call developer consuming logic(like reactive or callback).
call from hazelcast(client) to comsumer logic when add queue data.

receiver:
`queue.regItemConsumer(new ConsumeLogic());`

like ItemListener registration  but poll data only one client and registration only one consumer logic same hazelcast node or client
(other hazelcast node or client can registration difference consuming logic on same queue)
Hazelcast need queue's road balancing (ex: round robin)

thanks.

**Comment by mmedenjak** [comment]
Hi @ppsnumi !

If I understood correctly, you would like to add a listener which will be triggered and which will consume an item from the queue. If there are multiple listeners, only one should consume the queue item. If that is true, this is actually a duplicate of these issues:
https://github.com/hazelcast/hazelcast/issues/5462
https://github.com/hazelcast/hazelcast/issues/2783
https://github.com/hazelcast/hazelcast/issues/11819

This enhancement is best suited for `ITopic` and not `IQueue` since this is mostly a messaging pattern and even queues in JDK don't support this kind of notification mechanism. Closing this issue as a duplicate of the existing enhancement requests. 

Until this is implemented, maybe you can try a different approach - a combination of `ITopic` and `IQueue` - the producer can add an item to the queue and then send a message to the `ITopic`. The consumers can listen on the `ITopic` and once they receive a message, they can call `IQueue#poll` to fetch the queue item. All of the listeners will receive the message but only one will be able to poll the queue. That sounds like it will achieve what you wanted, without waiting on us to implement the feature.
